lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1130,11 +1130,14 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
               mac_pkg_name = "puppet-agent-#{opts[:puppet_agent_version]}"
               version = version[0,2] + '.' + version[2,2] if (variant =~ /osx/ && !version.include?("."))
               path_chunk = ''
-              # new hotness
+              # newest hotness
               path_chunk = "apple/#{version}/#{opts[:puppet_collection]}/#{arch}"
               release_path << path_chunk
-              release_file = "#{mac_pkg_name}-1.#{codename}.dmg"
-              if not link_exists?("#{release_path}/") # oops, try the old stuff
+              release_file = "#{mac_pkg_name}-1.osx#{version}.dmg"
```

Comment:
Could you clear up the code here with a comment indicating the two (three?) possible file name versions and how you are sifting through them?